macros/union_data.sql

```diff
@@ -22,67 +22,94 @@
     default_variable,
     union_schema_variable,
     union_database_variable
-    ) %}
+    ) -%}
 
-{% if var(union_schema_variable, none) %}
+{%- if var(union_schema_variable, none) -%}
 
-    {% set relations = [] %}
+    {%- set relations = [] -%}
     
-    {% if var(union_schema_variable) is string %}
-    {% set trimmed = var(union_schema_variable)|trim('[')|trim(']') %}
-    {% set schemas = trimmed.split(',')|map('trim'," ")|map('trim','"')|map('trim',"'") %}
-    {% else %}
-    {% set schemas = var(union_schema_variable) %}
-    {% endif %}
-
-    {% for schema in var(union_schema_variable) %}
-    {% set relation=adapter.get_relation(
+    {%- if var(union_schema_variable) is string -%}
+    {%- set trimmed = var(union_schema_variable)|trim('[')|trim(']') -%}
+    {%- set schemas = trimmed.split(',')|map('trim'," ")|map('trim','"')|map('trim',"'") -%}
+    {%- else -%}
+    {%- set schemas = var(union_schema_variable) -%}
+    {%- endif -%}
+
+    {%- for schema in var(union_schema_variable) -%}
+    {%- set relation=adapter.get_relation(
         database=source(schema, table_identifier).database if var('has_defined_sources', false) else var(database_variable, default_database),
         schema=source(schema, table_identifier).schema if var('has_defined_sources', false) else schema,
         identifier=source(schema, table_identifier).identifier if var('has_defined_sources', false) else table_identifier
     ) -%}
     
-    {% set relation_exists=relation is not none %}
+    {%- set relation_exists=relation is not none -%}
 
-    {% if relation_exists %}
+    {%- if relation_exists -%}
+        {%- do relations.append(relation) -%}
+    {%- endif -%}
 
-    {% do relations.append(relation) %}
+    {%- endfor -%}
     
-    {% endif %}
-
-    {% endfor %}
-
-    {{ dbt_utils.union_relations(relations) }}
-
-{% elif var(union_database_variable, none) %}
-
-    {% set relations = [] %}
-
-    {% for database in var(union_database_variable) %}
-
-    {% set relation=adapter.get_relation(
+    {%- if relations != [] -%}
+        {{ dbt_utils.union_relations(relations) }}
+    {%- else -%}
+    {% if execute and not var('remove_empty_table_warnings', false) -%}
+    {{ exceptions.warn("\n\nPlease be aware: The " ~ table_identifier ~ " table was not found in your destination. The package will create a completely empty staging model as to not break downstream transformations. To turn off these warnings, set the `remove_empty_table_warnings` variable to False.\n") }}
```

Comment:
What are your thoughts on adding the package information to the warning? I tested this locally where I was doing this union behavior for three packages and found it to be quite confusing which package was doing what with the staging models.

I found we can add the following to add a bit more details.
```sql
{{ exceptions.warn("\n\nPlease be aware: The " ~ table_identifier ~ " table was not found in your destination. The " ~ default_schema ~ " package will create a completely empty staging model as to not break downstream transformations. To turn off these warnings, set the `remove_empty_table_warnings` variable to False.\n") }}
```

Results:
```zsh
Please be aware: The discount_code table was not found in your destination. The shopify package will create a completely empty staging model as to not break downstream transformations. To turn off these warnings, set the `remove_empty_table_warnings` variable to False.

**AND**

Please be aware: The user_history table was not found in your destination. The instagram_business_pages package will create a completely empty staging model as to not break downstream transformations. To turn off these warnings, set the `remove_empty_table_warnings` variable to False.
```

It isn't fool proof as the naming won't match exactly what we have named the package (ie. we don't call the instagram_business package instagram_business_pages), but it may be enough to add more context.

I am trying to future proof this and I foresee us using this across a lot more packages, and feel this could get overwhelming for customers if they are unaware where these tables are being used. Thoughts?

Reply:
lmk whatcha think about what i added!

Reply:
Looks great, I like the added details to the warning message!